### ISSUE TITLE
Extension host terminated unexpectedly -- also with "--disable-extensions"

### ISSUE THREAD

**Issue by cbreit** [microsoft/vscode#88012]
### Summary
VS code is unable to start the extension host -- also when disabling extensions or deleting all of them manually.
A reinstall or removal of the settings folder did not affect the behavior. Same problem across different users and window managers (KDE/Xfce).

### Console output:
```
abstractExtensionService.ts:155 Extension host terminated unexpectedly. Code:  null  Signal:  SIGILL
  | _onExtensionHostCrashed | @ | abstractExtensionService.ts:155
  | _onExtensionHostCrashed | @ | extensionService.ts:375
  | _onExtensionHostCrashOrExit | @ | abstractExtensionService.ts:147
  | (anonymous) | @ | abstractExtensionService.ts:137
  | fire | @ | event.ts:581
  | _onExtHostProcessExit | @ | extensionHost.ts:469
  | (anonymous) | @ | extensionHost.ts:218
  | emit | @ | events.js:200
  | ChildProcess._handle.onexit | @ | internal/child_process.js:272

notificationsAlerts.ts:40 Extension host terminated unexpectedly.
  | onDidNotificationChange | @ | notificationsAlerts.ts:40
  | (anonymous) | @ | notificationsAlerts.ts:26
  | fire | @ | event.ts:581
  | addNotification | @ | notifications.ts:171
  | notify | @ | notificationService.ts:101
  | prompt | @ | notificationService.ts:166
  | _onExtensionHostCrashed | @ | extensionService.ts:395
  | _onExtensionHostCrashOrExit | @ | abstractExtensionService.ts:147
  | (anonymous) | @ | abstractExtensionService.ts:137
  | fire | @ | event.ts:581
  | _onExtHostProcessExit | @ | extensionHost.ts:469
  | (anonymous) | @ | extensionHost.ts:218
  | emit | @ | events.js:200
  | ChildProcess._handle.onexit | @ | internal/child_process.js:272

dom.ts:216 [Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

event.ts:19 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

extensionHost.ts:203 Extension Host
extensionHost.ts:204 
#
# Fatal error in , line 0
# ignored
#
#FailureMessage Object: 0x7ffd8c886e80
```

Unfortunately VS Code logs do not contain any useful information; there's also no "exthost" file.

### System environment
Version: 1.41.1
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2019-12-18T15:04:31.999Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Linux x64 5.3.0-3-amd64 -- Debian sid

**Comment by ChaseKnowlden** [comment]
Kamino cloned this issue to [ChaseKnowlden/vscode](https://github.com/ChaseKnowlden/vscode/issues/2)

**Comment by cbreit** [comment]
**Update:**
Extension host **works** up to 1.39.2
Version: 1.39.2
Commit: 6ab598523be7a800d7f3eb4d92d7ab9a66069390
Date: 2019-10-15T15:33:40.634Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 5.4.0-2-amd64

Extension host start to **crash from 1.40 onwards**
Version: 1.40.0
Commit: 86405ea23e3937316009fc27c9361deee66ffbf5
Date: 2019-11-06T18:14:08.920Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Linux x64 5.4.0-2-amd64

**Comment by schattenmann80** [comment]
I think I have the same problem. I use Debian.

I tryed "-disable-extensions"

![grafik](https://user-images.githubusercontent.com/22788790/72718262-e48ed600-3b75-11ea-90d3-44c2e9c360a2.png)

![Unbenannt](https://user-images.githubusercontent.com/22788790/72718300-fd978700-3b75-11ea-8189-fbc2452031ea.PNG)

**Comment by alexdima** [comment]
@deepak1556 Is there a way to learn more about the crash on Linux? It seems to first happen in 1.40.0, with the Electron 6 update.

@cbreit @schattenmann80 Have you tried VS Code insiders? VS Code insiders is now on Electron 7.

**Comment by schattenmann80** [comment]
@alexdima No I haven't tried VS Code insiders yet. I let you know when I have time to try it.

**Comment by deepak1556** [comment]
@alexdima we could pause the extension host with `--inspect-brk` and attach gdb to the process for a better stack trace.

**Comment by schattenmann80** [comment]
I tried it with: Version: 1.42.0-insider
Commit: 33c79d5ad447956814a2a3658029dffb9e28bae6
Date: 2020-01-15T08:55:46.564Z
Electron: 7.1.7
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 4.9.0-9-amd64

The problem is still there.

**Comment by cbreit** [comment]
@alexdima The issue still exists with the insiders build

Version: 1.42.0-insider
Commit: ee803826ad8f16534bbf72fa463306ebcb1e7ee6
Date: 2020-01-22T14:43:37.619Z
Electron: 7.1.7
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.4.0-3-amd64

I'm running Debian sid also in 2 VBox VMs -- interestingly it's working in these VMs. Already thought that it could then be GPU related (both workstations are on Nvidia with the proprietary driver -- however a "--disable-gpu" had no effect; also not on the "insiders" build.

Tried debugging with gdb and chrome-inspect; however, there are a couple of exceptions caught in the dev-tools during startup:
* internal/bootstrap/loaders.js  line 118
`mod = bindingObj[module] = getLinkedBinding(module);`
`"No such module was linked: atom_browser_v8_util"`
* same error at the same function for: 
`atom_browser_features, atom_browser_command_line, atom_browser_clipboard, atom_browser_features, atom_browser_native_image, atom_browser_screen, atom_browser_crash_reporter
`
* internal/modules/cjs/loader.js – line 720
`Module._resolveFilename = function(request, parent, isMain, options)`
`"Cannot find module '/usr/share/code-insiders/resources/app/node_modules.asar/spdlog/build/spdlog.node' Require stack: - /usr/share/code-insiders/resources/app/node_modules.asar/spdlog/node_modules/bindings/bindings.js - /usr/share/code-insiders/resources/app/node_modules.asar/spdlog/index.js - /usr/share/code-insiders/resources/app/out/bootstrap-amd.js - /usr/share/code-insiders/resources/app/out/main.js"`

Any specific ones to explicitly dig in deeper?
Nothings sounded like being related to the extension-host.

gdb did not show interesting stuff -- just a gtk warning, threads being started and an electron deprecation warning.

**Comment by schattenmann80** [comment]
@cbreit @alexdima that it is a GPU related issue is a possible option, because I am working on a Debian workstation with no GPU where I connect via Remotedesktop.

**Comment by alexdima** [comment]
@cbreit @schattenmann80 Could you please try the following:
* download VS Code as a `.tar.gz` from https://code.visualstudio.com/insiders/#linux
* rename `$HOME/.config/Code-Insiders/` (if you have it) to something else so we get a fresh start
* unpack in a folder somewhere, be sure to have all permissions set to the current user and not root or anything like that
* go to where you have unpacked vscode-insiders and look for the `resources/app/extensions` folder and rename it to `extensions1` and create a new `extensions` folder. 
* this should make it that vs code runs without any 3rd party and without any built-in extensions

Does the extension host still crash?

**Comment by schattenmann80** [comment]
> 
> 
> @cbreit @schattenmann80 Could you please try the following:
> 
>     * download VS Code as a `.tar.gz` from https://code.visualstudio.com/insiders/#linux
> 
>     * rename `$HOME/.config/Code-Insiders/` (if you have it) to something else so we get a fresh start
> 
>     * unpack in a folder somewhere, be sure to have all permissions set to the current user and not root or anything like that
> 
>     * go to where you have unpacked vscode-insiders and look for the `resources/app/extensions` folder and rename it to `extensions1` and create a new `extensions` folder.
> 
>     * this should make it that vs code runs without any 3rd party and without any built-in extensions
> 
> 
> Does the extension host still crash?

It is working.
No extension host crash anymore. :)

But I can't work without extensions :(

**Comment by alexdima** [comment]
Ok, that is interesting. This indicates to me that a built-in extension is causing the crash. Does the crash happen all the time for you or only when opening certain file types?

Out of the box, the following extensions activate on `*` (always):
* `debug-auto-launch`
* `emmet`
* `git`
* `merge-conflict`
* `search-result`
* `vscode-account`

Could you try to bring back all the extension except the ones I listed above, and perhaps move the ones above back one-by-one, until we find out which one is the likeliest culprit.

By the way, don't you have a log at `F1 > Developer: Show logs... > Extension Host` ?

**Comment by schattenmann80** [comment]
I found something. I use a script to set up my shell environment and if I start vscode without running the script all is working fine. But If I use the script before the start of vscode the extenstion host is crashes.

If I use the script vscode-insider is still crashing. Also with the extansions folder empty.

I try to look into your envirment setup script.

**Comment by alexdima** [comment]
@schattenmann80 Please share with us the script (if possible), maybe there is something in the ENV that causes us to crash?

**Comment by cbreit** [comment]
@alexdima Deleting the extensions still causes a crash on my side (I've started vscode with --no-sandbox -- hope that does not cause any side effects).
The extensions.log file in the "output-<someID>" folder had 0 bytes. Tried it with the .tar.gz file and the .deb file (removed the extensions in /usr/share/code-insiders/resources/app)

However, the env-variables are very hot (thx @schattenmann80 ) -- starting vscode (1.41 or insiders-build) in a clean environment works.

The LANG env-variable causes the extension-host to crash -- mine is set to:
LANG=de_AT@euro

Unsetting it and everything works like a charm. It explicitely seems to have problems with the "@euro" locale -- using UTF-8 causes no problems.

Hope this helps to identify the underlying issue.
Thx @alexdima for your help.

**Comment by schattenmann80** [comment]
@alexdima I tested the LANG env-variables and I have the same Problem as @cbreit 

My env-setup-script sets:
LANG=de_DE@euro
LANGUAGE=de_DE@euro
LC_CTYPE="de_DE@euro"
LC_NUMERIC="de_DE@euro"
LC_TIME="de_DE@euro"
LC_COLLATE="de_DE@euro"
LC_MONETARY="de_DE@euro"
LC_MESSAGES="de_DE@euro"
LC_PAPER="de_DE@euro"
LC_NAME="de_DE@euro"
LC_ADDRESS="de_DE@euro"
LC_TELEPHONE="de_DE@euro"
LC_MEASUREMENT="de_DE@euro"
LC_IDENTIFICATION="de_DE@euro"
LC_ALL=de_DE@euro

After I set manually to:
LANG=de_DE.UTF-8
LANGUAGE=
LC_CTYPE=de_DE.UTF-8
LC_NUMERIC=de_DE@euro
LC_TIME=de_DE.UTF-8
LC_COLLATE=de_DE.UTF-8
LC_MONETARY=de_DE.UTF-8
LC_MESSAGES=de_DE.UTF-8
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL=

vscode is running fine.

**Comment by deepak1556** [comment]
Thanks @cbreit @schattenmann80 I am able to repro the issue setting the LANG env variable.

**Comment by deepak1556** [comment]
@alexdima this is same as https://github.com/microsoft/vscode/issues/84803 but the fix that landed didn't cover this use case. We need to backport https://unicode-org.atlassian.net/browse/ICU-20558 icu patch in electron, will look into this.

**Comment by alexdima** [comment]
@cbreit @schattenmann80 Thank you for your patience and helping us find the root cause 🎉 🎉 🎉

@deepak1556 Thank you, I will let you drive this

**Comment by deepak1556** [comment]
The fix is available in our latest stable `1.43`.
